tests/Feature/GroupTest.php

```diff
@@ -0,0 +1,181 @@
+    public function an_admin_can_see_a_course_groups()
+    {
+        $admin = factory(User::class)->states('admin')->create();
+        $course = factory(Course::class, 20)->create([
+            'group_min' => 3,
+            'group_max' => 3,
+        ])->first();
+
+        $response = $this->actingAs($admin)
+            ->get(route('admin.groups.show', $course->id));
+
+        $response->assertStatus(200);
+        $response->assertViewHas('groups');
+    }
+
+    /** @test */
+    public function an_admin_can_see_a_course_index_with_groups_size()
+    {
+        $admin = factory(User::class)->states('admin')->create();
+        factory(Course::class, 20)->create([
+            'group_min' => 3,
+            'group_max' => 3,
+        ]);
+
+        $response = $this->actingAs($admin)
+            ->get(route('admin.groups.index'));
+
+        $response->assertStatus(200);
+        $response->assertViewHas('courses');
+    }
+
+    /** @test */
+    public function a_student_can_see_his_group_in_a_course()
+    {
+        $student = factory(Student::class)->create();
+        $course = factory(Course::class, 2)->create([
+            'group_min' => 1,
+            'group_max' => 1,
+        ])->first();
+        $group = factory(Group::class)->create([
+            'course_id' => $course->id,
+        ]);
+
+        $student->join($group);
+
+        $response = $this->actingAs($student->user)
+            ->get(route('groups.show', $course->id));
+
+        $response->assertStatus(200);
+        $response->assertViewHas('membership');
+    }
+
+    /** @test */
+    public function a_student_can_accept_a_group_invitation()
+    {
+        $student = factory(Student::class)->create();
+        $course = factory(Course::class)->create([
+            'group_min' => 2,
+            'group_max' => 2,
+        ]);
+        $group = factory(Group::class)->create([
+            'course_id' => $course->id,
+        ]);
+
+        $invitation = factory(Invitation::class)->create([
+            'student_number' => $student->student_number,
+            'course_id' => $course->id,
+            'group_id' => $group->id,
+        ]);
+
+        $response = $this->actingAs($student->user)
+            ->get(route('groups.update', $invitation->id));
+
+        $this->assertEquals($group->id, $student->memberships()->first()
+            ->group()->first()->id);
+    }
+
+    /** @test */
+    public function a_student_cant_accept_invalid_invitation()
+    {
+        $invitationCount = Invitation::count();
+
+        $student = factory(Student::class)->create();
+        $course = factory(Course::class)->create();
+        $group = factory(Group::class)->create();
+
+        $invitation = factory(Invitation::class)->create([
+            'student_number' => factory(Student::class)->create()->student_number,
+            'course_id' => $course->id,
+            'group_id' => $group->id,
+        ]);
+
+        $this->actingAs($student->user)
```

Comment:
Should the assert that the response had the status code 302 (redirect)

Reply:
👍